### PR TITLE
ci: Test for unused dependencies

### PR DESCRIPTION
**Pull request by JonathanWoollett-Light**
## Changes

Adds a test for unused cargo dependencies.

## Reason

Prevents including unused dependencies see https://github.com/firecracker-microvm/firecracker/pull/4113#discussion_r1331364257.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] API changes follow the [Runbook for Firecracker API changes][2].
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- [x] New `TODO`s link to an issue.
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Comment by JonathanWoollett-Light** [comment]
Blocked on docker changes from: https://github.com/firecracker-microvm/firecracker/pull/4089

**Comment by roypat** [comment]
Unblocked :)

**Comment by JonathanWoollett-Light** [comment]
https://github.com/firecracker-microvm/firecracker/pull/4089 merged, unblocked.

**Review by roypat — APPROVED** [review]

**Review by bchalios — CHANGES_REQUESTED** [review]
Could you please break this down to two commits, one that adds the test (test, Docker changes) and one that removes the unused dependencies?

Also, there are here, seemingly, completely unrelated changes (the ones in the documentation). Please drop these and add them in a different PR.

**Comment by JonathanWoollett-Light** [comment]
> Could you please break this down to two commits, one that adds the test (test, Docker changes) and one that removes the unused dependencies?

I've updated the PR breaking it down into 4 commits.

> Also, there are here, seemingly, completely unrelated changes (the ones in the documentation). Please drop these and add them in a different PR.

These are required by the updated version of markdown lint. `mdl` is a ruby dependency so we don't have a lock file for it.

**Review by roypat — DISMISSED** [review]

File: tests/integration_tests/build/test_dependencies.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (pb8o) [inline tests/integration_tests/build/test_dependencies.py:1]: 2024 :)

**Review by pb8o — APPROVED** [review]

File: tests/integration_tests/build/test_dependencies.py (hunk 1 above)
Comment (roypat) [reply tests/integration_tests/build/test_dependencies.py]: LOL

fixed

**Review by roypat — DISMISSED** [review]

**Review by roypat — APPROVED** [review]

**Pull request merged**